CrowdStrike/crowdstrike_telemetry/pull_telemetry_events.py

```diff
@@ -123,8 +144,24 @@ async def process_s3_file(self, key: str, bucket: str | None = None) -> list[str
             if content[0:2] == b"\x1f\x8b":
                 logger.info(f"Decompressing file by key {key}")
                 result_content = decompress(content)
-
-        return [line for line in result_content.decode("utf-8").split("\n") if len(line.strip()) > 0]
+        result = []
+        for line in result_content.decode("utf-8").split("\n"):
+            if len(line.strip()) > 0:
+                try:
+                    event = json.loads(line)
+                    if event.get("event_simpleName") is None or event.get("event_simpleName") in EXCLUDED_EVENT_ACTIONS:
+                        DISCARDED_EVENTS.labels(intake_key=self.configuration.intake_key).inc()
+                        continue
+                    result.append(line)
+                except Exception as any_exception:
+                    logger.error(
+                        "failed to read line from event stream",
+                        line=line,
+                        stream_root_url=self.stream_root_url,
+                    )
+                    self.log_exception(any_exception)
+                    raise any_exception
```

Comment:
I'm not sure if we should raise an exception here if we failed to decode the event.
The previous code didn't care if the content was a JSON-document. So, I suggest logging the exception but not raising the exception (that will stop here the processing of the file).